cmd/store/export.go

```diff
@@ -0,0 +1,169 @@
+		storeName = store.Name
+	}
+
+	// get the model
+	model, err := authorizationmodel.ReadFromStore(config, fgaClient)
+	if err != nil {
+		return nil, err
+	}
+
+	authModel := authorizationmodel.AuthzModel{}
+	authModel.Set(*model.AuthorizationModel)
+
+	dsl, err := authModel.DisplayAsDSL([]string{"model"})
+	if err != nil {
+		return nil, fmt.Errorf("unable to get model dsl: %w", err)
+	}
+
+	// get the tuples
+	rawTuples, err := tuple.Read(fgaClient, &client.ClientReadRequest{}, 10)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read tuples: %w", err)
+	}
+
+	var tuples []client.ClientContextualTupleKey
+	for _, t := range rawTuples.GetTuples() {
+		tuples = append(tuples, t.GetKey())
+	}
+
+	// get the assertions
+	assertionOptions := client.ClientReadAssertionsOptions{
+		AuthorizationModelId: authModel.ID,
+	}
+
+	assertionResponse, err := fgaClient.ReadAssertions(context.Background()).Options(assertionOptions).Execute()
+	if err != nil {
+		return nil, fmt.Errorf("unable to read assertions: %w", err)
+	}
+
+	assertions := assertionResponse.GetAssertions()
+	modelChecks := map[string]storetest.ModelTestCheck{}
+
+	for _, assertion := range assertions {
+		key := fmt.Sprintf("%s|%s", assertion.TupleKey.User, assertion.TupleKey.Object)
+		_, exists := modelChecks[key]
+
+		if !exists {
+			modelChecks[key] = storetest.ModelTestCheck{
+				User:       assertion.TupleKey.User,
+				Object:     assertion.TupleKey.Object,
+				Assertions: map[string]bool{},
+			}
+		}
+
+		modelChecks[key].Assertions[assertion.GetTupleKey().Relation] = assertion.Expectation
+	}
+
+	var checks []storetest.ModelTestCheck
+	for _, value := range modelChecks {
+		checks = append(checks, value)
+	}
+
+	storeData := &storetest.StoreData{
+		Name:   storeName,
+		Model:  *dsl,
+		Tuples: tuples,
+		Tests: []storetest.ModelTest{
+			{
+				Name:  "Tests",
+				Check: checks,
+			},
+		},
+	}
+
+	return storeData, nil
+}
+
+// exportCmd represents the export store command
+var exportCmd = &cobra.Command{
+	Use:     "export",
+	Short:   "Export store data",
+	Long:    `Export a store to YAML`,
+	Example: "fga store export",
+	RunE: func(cmd *cobra.Command, _ []string) error {
+		clientConfig := cmdutils.GetClientConfig(cmd)
+
+		fgaClient, err := clientConfig.GetFgaClient()
+		if err != nil {
+			return fmt.Errorf("failed to initialize FGA Client due to %w", err)
+		}
+
+		storeData, err := buildStoreData(clientConfig, fgaClient)
+		if err != nil {
+			return fmt.Errorf("failed to export store: %w", err)
+		}
+
+		if storeData != nil {
+			storeYaml, err := yaml.Marshal(storeData)
+			if err != nil {
+				return fmt.Errorf("unable to marshal storedata yaml: %w", err)
+			}
+
+			fileName, _ := cmd.Flags().GetString("output-file")
+			if fileName == "" {
+				fmt.Println(string(storeYaml))
+				return nil
+			}
+
+			err = os.WriteFile(fileName, storeYaml, 0666)
+			if err != nil {
+				return err
+			}
+		}
+
+		return output.Display(output.EmptyStruct{})
```

Comment:
Is this the correct response to return here when writing to a file?

Reply:
AFAIK we don't really have a guideline for output (i.e. should it always be in a machine parseable format like json or can we output normal text). I would maybe tend to output a confirmation like `wrote store to file <filename>` but maybe we can revisit that alongside general output?

Reply:
I added a message here to say it'd been written to a file, seemed more friendly in the light of adding a confirmation prompt.

Reply:
Although actually it writes the message as well as `{}` - should it just print the message only?

Reply:
That's fine to me, I think for a command like this it's potentially expected to have multiple logs output throughout the time it runs